### ISSUE TITLE
MCC Multilabel with average macro

### ISSUE THREAD

**Issue by Buzzeitor30** [Lightning-AI/torchmetrics#3363]
## 🚀 Feature
The current behaviour of the multilabel MCC is to calculate the overall MCC by adding the TP, TN, FP & FN of each label (C matrices of shape (2,2)). This does, however, lead to an unintended behaviour in which the metric is computed as a micro average and never as macro.

### Motivation
There are disbalanced multilabel classification tasks which might rely on the MCC as their primary metric. By not offering the option to select the average on this metric, this often leads to a not representative metric regarding the problem.

### Pitch
Add the possibility to compute macro average on MCC.

### Alternatives
I've done this implementation for my code, although it is a bit messy, but to give an idea of what we are looking for
```python
class MacroMCCFromBinary(Metric):
    """Macro-averaged MCC: mean of per-label binary MCCs (mirrors STL heads)."""

    def __init__(self, num_labels: int):
        super().__init__()
        self.num_labels = num_labels
        self._binary_mccs = torch.nn.ModuleList(
            [MatthewsCorrCoef(task="binary") for _ in range(num_labels)]
        )

    def update(self, preds: torch.Tensor, target: torch.Tensor) -> None:
        for i, mcc in enumerate(self._binary_mccs):
            mcc.update(preds[:, i], target[:, i])

    def compute(self) -> torch.Tensor:
        return torch.stack([mcc.compute() for mcc in self._binary_mccs]).mean()

    def reset(self) -> None:
        for mcc in self._binary_mccs:
            mcc.reset()
```

### Additional context

I think part of the fix is located on https://github.com/Lightning-AI/torchmetrics/blob/master/src/torchmetrics/functional/classification/matthews_corrcoef.py

There must be some kind of fix.

**Comment by Ashleyyq** [comment]
Hi @Buzzeitor30,  I would like to take this one if it is still open.
My proposed approach: add an average="macro" branch to the multilabel MCC, computing MCC independently per label from each label TP/TN/FP/FN, then averaging across labels, rather than the current behavior of summing the confusion entries across labels first. I would keep the average argument style consistent with the other classification metrics in the library. For correctness I plan to align the numerical results against sklearn matthews_corrcoef computed per label, and add unit tests covering the multilabel macro case.
I also noticed MCC reduction is being actively touched in #3355 and #3384, so I will make sure to rebase on the latest reduction logic before opening the PR.
Does this direction look right to you, or would you prefer a different API shape for average here? Happy to adjust before I start.

**Comment by seaic-mac-murchadha** [comment]
Hi @Buzzeitor30 @Ashleyyq ,

Is this currently in progress? If not, I'm happy to take ownership.
